src/wallet/wallet.cpp

```diff
@@ -3618,14 +3714,41 @@ CWallet* CWallet::CreateWalletFromFile(const std::string walletFile)
         walletInstance->SetMaxVersion(nMaxVersion);
     }
 
+
+    std::string hdWatchOnly = GetArg("-hdwatchonly", "");
+    CExtPubKey extPubKey;
+    if (!hdWatchOnly.empty()) {
+        CBitcoinExtPubKey bitcoinExtPubKey;
+        auto extpubKeyPrefix = chainparams.Base58Prefix(CChainParams::Base58Type::EXT_PUBLIC_KEY);
+        if (!bitcoinExtPubKey.SetString(hdWatchOnly.c_str(), extpubKeyPrefix.size())) {
+            InitError(_("Invalid ExtPubKey format"));
+            return NULL;
+        }
+        extPubKey = bitcoinExtPubKey.GetKey();
+    }
+
+    if (!fFirstRun && !hdWatchOnly.empty()) {
```

Comment:
I got no warning when I started bitcoind again with a different `-hdwatchonly=xpub`.
Here there should be a check wether the wallet has already a watch-only xpub set, and if the user provides one, and it's different, it should warn or stop.